### ISSUE TITLE
Removed default language folder

### ISSUE THREAD

**Issue by Faulik** [vinissimus/next-translate#147]
This commit line was breaking change:
https://github.com/vinissimus/next-translate/blob/master/cli/builder.js#L57

this folder disappeared:
![image](https://user-images.githubusercontent.com/749225/80716356-31346680-8b00-11ea-92ed-561e110c7f38.png)

if one is using this config from example:
```js
{
  "allLanguages": ["en", "ca", "es"],
  "defaultLanguage": "en",
  "currentPagesDir": "pages_",
  "finalPagesDir": "pages",
  "localesPath": "locales",
  "pages": {
    "*": ["common"],
    "/": ["home", "example"],
    "/about": ["about"]
  }
}
```

There should be a redirect if we want to make fewer files or return folder to keep it without breaking change.

**Comment by aralroca** [comment]
This was considered as a bug. Fixed on `0.15.0`, on https://github.com/vinissimus/next-translate/pull/123.

It can look like a breaking change but is changing a wrong behavior.

The correct behavior is:

* `redirectToDefaultLang=false` (default): Is displaying the current language as `/` instead of `/en`. Navigating directly to `/en` in this case is returning a 404 because is not a valid URL. 
* `redirectToDefaultLang=true`: Is displaying the current language as `/en`. Navigating directly to `/` is redirecting to the default lang `/en`.

Before, with `redirectToDefaultLang=false` you had duplicated pages, and it was possible to access to `/` and `/en` rendering the same page. This was considered as a wrong behavior, and fixed.

**Comment by Faulik** [comment]
Is there an option to add a redirect from `/en` to `/` for default language? Maybe the same way as `redirectToDefaultLang=true` does it.
Also, we should update readme to reflect new behaviour.

**Comment by aralroca** [comment]
@Faulik the README is saying the current behavior:

> When it's set to `true`, the route `/some-page` redirects to `/en/some-path` (if `en` is the default language). When it's set to `false`, entering to `/some-path` renders the page with the default language but without redirecting.     

However, maybe we can clarify that with `redirectToDefaultLang=false` is displaying a 404 page accessing directly to `/en` (if English is de default language). 

> Is there an option to add a redirect from /en to / for default language? Maybe the same way as redirectToDefaultLang=true does it.

Using the `next-translate` CLI there isn't a way to do this (at least right now)... But for now, there are some options that maybe you can try to work-around:

* Create manually a Page under `/en/[...catchAll].js` that redirects to `/` in the case that the pathname starts with `/en`, and you render the 404 page in other cases (to avoid `/es/en`...).
* Similar that the first point. But you always redirect to `/`,and create a script to remove the nested `/en` folder generated inside other languages.  So instead of `next-translate && next build`, you will need to modify to `next-translate && ./my-custom-script.sh && next build`. In this case, you are going to get a real 404 status code on `/es/en` cases...
* Using Customs Routes with `redirect` (is experimental in next.js) https://github.com/zeit/next.js/issues/9081

Hope that it helps.

Nevertheless, it can be interesting to support this in our CLI.

**Comment by Faulik** [comment]
Readme in parts that I quoted, as there are example structure and example config for that structure and they don't match.

> Nevertheless, it can be interesting to support this in our CLI.
I think first option will be pretty straightforward and logical. Should I create PR for this?

**Comment by aralroca** [comment]
Feel free to PR 🙂 ! If you have doubts you can create a Draft PR and I can help you.

**Comment by aralroca** [comment]
This will be released on `0.16.0`. Currently it's in `0.16.0-canary.1`

**Comment by aralroca** [comment]
I reopen the issue because this is breaking a real project that have the `i18nMiddleware` with `redirectToDefaultLang=false` and another middleware to manage all the i18n redirects, for example, to redirect the languages to other domains. Ex: https://www.vinissimus.com/de/ is redirecting to https://www.hispavinus.de/de/

So we need to keep the possibility to don't use any redirect.

I propose the next for backward compatibility:

* Deprecate `redirectToDefaultLang` but keeping the same usage from `0.15.0` until `1.0.0` version. Should be removed in `1.0.0`
* New property `defaultLangRedirect`, with this options: `root`, `lang-path`, `none'` And `none` as default.

@Faulik if you agree I can implement this.

**Comment by Faulik** [comment]
Previous behaviour was also breaking a real project.

Sure, we can have it under different option if that makes sense.

**Comment by aralroca** [comment]
@Faulik the new `defaultLangRedirect` will be available on `0.16.0` (currently in `0.16.0-canary.4`).
